apps/extension/server/src/server.ts

```diff
@@ -215,23 +238,43 @@ async function handleDocumentChange(textDocument: TextDocument): Promise<void> {
     });
 
     if (displayedMachine?.uri === textDocument.uri) {
-      const machineResult = machineResults[displayedMachine.machineIndex]!;
-      if (
-        !deepEqual(
-          previouslyCachedDocument!.machineResults[
-            displayedMachine.machineIndex
-          ]!.toConfig({ anonymizeInlineImplementations: true }),
-          machineResult.toConfig({ anonymizeInlineImplementations: true }),
-        )
-      ) {
-        connection.sendNotification('displayedMachineUpdated', {
-          config: machineResult.toConfig()!,
-          layoutString: machineResult.getLayoutComment()?.value,
-          implementations: getInlineImplementations(machineResult, text),
-          namedGuards: machineResult
-            .getAllConds(['named'])
-            .map((elem) => elem.name),
+      const { configError, machineResult } =
+        extractionResults[displayedMachine.machineIndex];
+
+      if (configError) {
+        connection.sendNotification('extractionError', {
+          message: configError,
         });
+      } else {
+        // If we got this far we can safely assume that the machine config is valid and we can clear any potential errors
+        connection.sendNotification('extractionError', {
+          message: undefined,
+        });
```

Comment:
this again got a little bit "chatty"/noisy. You had a good filtering in place here:
https://github.com/statelyai/xstate-tools/blob/cd9dfcb579b20649e5eb069814df4d7f9ad0c3a0/apps/extension/server/src/server.ts#L238-L243

but that got lost after making the errors more local. I propose to keep `extractResults` in the `CachedDocument` to always get all the information that we need about the previous document. This would replace `CachedDocument['machineResults']` as those machine results are already part of the `extractResults`

Reply:
Good catch 🎣 I've just committed a less chatty version